### DISCUSSION TITLE
Modal Blade component has a problem with Nesting Components

### DISCUSSION THREAD

**Discussion by shebaoting** [filamentphp/filament#12530] [Help]
### Package

Other

### Package Version

v3.0.0

### How can we help you?

parent component
```
@foreach ($opportunitys as $opportunity)
    <livewire:company.opportunity.item :$opportunity :key="$opportunity->id" />
    @endforeach
```
child component
```
<x-filament::modal id="show-lost">

<form wire:submit="onLost">
    <x-filament::button type="submit" class="float-right">保存</x-filament::button>
</form>
</x-filament::modal>
```

```
class Item extends Component
{
    public $opportunity;
    public $opportunityId;
    public function render()
    {
        return view('livewire.company.opportunity.item');
    }


    public function showLost($id){
        $this->opportunityId = $id;
        $this->dispatch('open-modal', id: 'show-lost');
    }

    public function onLost()
    {
        dd($this->opportunityId);
    }

}
```

In showLost(), the value of $this->opportunityId is available, but the value of $this->opportunityId is not available in onLost(), it is null

If Nesting Components is not used. then everything is fine.

May I ask how this problem can be solved? Is it a bug?

**Comment by xalabama** [comment]
May be its because you don't have defined the property opportinityId, so livewire can't hold your data and your missing property returns a null, because it doesn't exists. 

Add the property like this should solve your problem.

```
class Item extends Component
{
    public $opportunity;
    
    // The missing property
    public $opportunityId;

    // ...
}
```

**Reply by shebaoting** [reply]
When I asked the question, I left out $opportunityId. in fact, $opportunityId is defined in the actual code. The problem still exists.

I'm sorry, I've reworked my question.
